posthog-react-native/src/surveys/components/SurveyModal.tsx

```diff
@@ -0,0 +1,103 @@
+
+import { Cancel } from './Cancel'
+import { ConfirmationMessage } from './ConfirmationMessage'
+import { Questions } from './Surveys'
+
+import { SurveyAppearanceTheme } from '../surveys-utils'
+import { Survey } from '../posthog-surveys-types'
+
+export type SlypModalProps = {
+  survey: Survey
+  appearance: SurveyAppearanceTheme
+  onShow: () => void
+  onClose: (submitted: boolean) => void
+}
+
+export function SurveyModal(props: SlypModalProps): JSX.Element | null {
+  const { survey, appearance, onShow } = props
+  const [isSurveySent, setIsSurveySent] = useState(false)
+  const onClose = useCallback(() => props.onClose(isSurveySent), [isSurveySent, props])
+
+  const surveyPopupDelayMilliseconds = appearance.surveyPopupDelaySeconds * 1000
+  const [isVisible, setIsVisible] = useState(surveyPopupDelayMilliseconds === 0)
+  if (surveyPopupDelayMilliseconds > 0) {
+    setTimeout(() => {
+      setIsVisible(true)
+    }, surveyPopupDelayMilliseconds)
+  }
+
+  const shouldShowConfirmation = isSurveySent && appearance.thankYouMessageHeader
+
+  useEffect(() => {
+    if (isVisible) {
+      onShow()
+    }
+  }, [isVisible, onShow])
+
+  useEffect(() => {
+    let timeout: NodeJS.Timeout | undefined
+    if (isVisible && shouldShowConfirmation && appearance.autoDisappear) {
+      timeout = setTimeout(() => {
+        onClose()
+      }, 5000)
+    }
+    return () => timeout && clearTimeout(timeout)
+  }, [isVisible, onClose, shouldShowConfirmation, appearance])
+
+  if (!isVisible) {
+    return null
+  }
+
+  return (
+    <Modal animationType="fade" transparent onRequestClose={onClose} statusBarTranslucent={true}>
+      <Pressable style={styles.modalContainer} onPress={onClose} accessible={false}>
+        <TouchableWithoutFeedback accessible={false}>
+          <View
+            style={[
+              styles.modalContent,
+              { borderColor: appearance.borderColor, backgroundColor: appearance.backgroundColor },
+            ]}
+          >
+            {!shouldShowConfirmation ? (
+              <Questions survey={survey} appearance={appearance} onSubmit={() => setIsSurveySent(true)} />
+            ) : (
+              <ConfirmationMessage
+                appearance={appearance}
+                header={appearance.thankYouMessageHeader}
+                description={appearance.thankYouMessageDescription}
+                contentType={appearance.thankYouMessageDescriptionContentType ?? 'text'}
+                onClose={onClose}
+                isModal={true}
+              />
+            )}
+            <View style={styles.topIconContainer}>
+              <Cancel onPress={onClose} appearance={appearance} />
+            </View>
+          </View>
+        </TouchableWithoutFeedback>
+      </Pressable>
+    </Modal>
+  )
+}
+
+const styles = StyleSheet.create({
+  modalContainer: {
+    flex: 1,
+    alignItems: 'center',
+    justifyContent: 'flex-end',
+    marginVertical: 40, // TODO Plus safe area?
```

Comment:
makes sense

Reply:
Okay I've added an "Optional" dependency on safe area insets like you've done with other libraries, but I'm not sure how I feel about it in this case.

The trouble is we're also dependent on the developer having a `<SafeAreaProvider>` outside where they put the `<PostHogProvider>`. I've put a fallback to some default insets if not, but it still seems like a hidden gotcha.

Reply:
we can document/add to code snippets, most people will just copy-pasta anyway